src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -161,6 +161,13 @@ private async Task WaitUntilHealthyAsync(IResource resource, IResource dependenc
             await WaitForResourceHealthyAsync(dependency.Name, cancellationToken).ConfigureAwait(false);
         }
 
+        // Now wait for the resource ready event to be executed.
+        resourceLogger.LogInformation("Waiting for resource ready to execute for '{Name}'.", dependency.Name);
+        resourceEvent = await WaitForResourceAsync(dependency.Name, re => re.Snapshot.ResourceReadyEventTask is { } t && t.IsCompleted, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
Does this work properly? I believe `WaitForResourceAsync` re-evaluates the predicate when the snapshot is published. The works when testing other data on the snapshot because it is immutable and requires a new snapshot to be published to update. But the task here is mutable. That means if `ResourceReadyEventTask` completes and there isn't a snapshot update afterwards then the wait doesn't complete.

Reply:
Would a fix be to update `FireResourceReadyEvent()` to await the eventing `PublishAsync` at the end of the method then do another notification publish?

Reply:
That's a good catch.

Reply:
I could instead wait until the task is non-null and then just use it.

Reply:
Updated, didn't keep IsComplete on the event, it's not needed 😄. I just check for a non-null event.